source/common/http/header_utility.cc

```diff
@@ -344,6 +344,15 @@ Http::Status HeaderUtility::checkRequiredHeaders(const Http::RequestHeaderMap& h
   return Http::okStatus();
 }
 
+Http::Status HeaderUtility::checkRequiredResponseHeaders(const Http::ResponseHeaderMap& headers) {
```

Comment:
can we have a code comment why this is here, rather than returning from the codec encode headers?

Reply:
done (https://github.com/envoyproxy/envoy/pull/16745/commits/46d285a2903de395ff768bc0b2dc20686b97720c)